pos_deposit/static/src/js/pos.js

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2021 Sunflower IT
+   License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). */
+
+odoo.define("pos_deposit.models", function(require) {
+
+    var models = require("point_of_sale.models");
+
+    var pos_model_super = models.PosModel.prototype;
+    models.PosModel = models.PosModel.extend({
+        initialize: function (session, attributes) {
+            /* Add the deposit related fields to the product model */
+            var self = this;
+            _.each(self.models, function(model) {
+                if (model.model === 'product.product') {
+                    model.fields.push('select_deposit');
+                    model.fields.push('use_deposit');
+                    domain_super = model.domain;
+                    model.domain = function(self) {
+                        domain = domain_super(self);
+                        domain.unshift(['is_deposit', '=', true]);
+                        domain.unshift('|');
+                        return domain;
+                    }
+                }
+            });
+            pos_model_super.initialize.apply(this, arguments);
+        }
+    });
+
+    var order_super = models.Order.prototype;
+    models.Order = models.Order.extend({
+        /* Create deposit line for products with a deposit. Check if
+           point_of_sale's add_product() has triggered a merge with an existing
+           line, and in that case don't do anything. Otherwise, add a line
+           with the deposit product. */
+        add_product: function (product, options) {
+            order_super.add_product.apply(this, arguments);
+
+            var merged_line = null;
+            this.get_orderlines().some(function (orderline) {
+                // TB: merged flag is set by us
+                if (orderline.merged) {
+                    merged_line = orderline;
+                    return true;
+                }
+            });
+            if (merged_line) {
+                merged_line.merged = false;
+            } else {
+                if (product.use_deposit) {
+                    if (product.select_deposit && product.select_deposit[0]) {
+                        var dep_prod = this.pos.db.get_product_by_id(product.select_deposit[0]);
+                        last = this.get_last_orderline();
+                        var dep_line = new models.Orderline({}, {
+                            pos: this.pos,
+                            order: this,
+                            product: dep_prod
+                        });
+                        this.add_orderline(dep_line);
+                        dep_line.deposit_for = last;
+                        last.deposit_line = dep_line;
+                    } else {
+                        alert ('Product for deposit is not configured properly !');
```

Comment:
Better to handle with aconstraint in pyhton. don't you think? It' avoid lazy error.

Reply:
We already have the constraint in the view, but not in Python. IMO it's enough in the view, what do you think